doc/classes/ShaderMaterial.xml

```diff
@@ -26,6 +26,7 @@
 			<description>
 				Changes the value set for this material of a uniform in the shader.
 				[b]Note:[/b] [param param] is case-sensitive and must match the name of the uniform in the code exactly (not the capitalized name in the inspector).
+				[b]Note:[/b] No error will be thrown if the type of [param value] does not match, and your shader may exhibit undefined behavior. This includes setting a GDScript [int]/[float] (which are 64-bit) into a shader int/float (which are 32 bit). This may lead to unintended consequences in cases where a high range or precision is required.
```

Comment:
This makes it more clear that casting does happen:

```suggestion
				[b]Note:[/b] No error will be thrown if the type of [param value] does not match. The value will be casted to the expected type silently, which may result in the shader exhibiting undefined behavior. This includes setting a GDScript [int]/[float] (which are 64-bit) into a shader int/float (which are 32 bit). This may lead to unintended consequences in cases where a high range or precision is required.
```